src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -177,1327 +205,1182 @@ static std::vector<rct_widget> _window_editor_object_selection_widgets = {
-{
-    *height = static_cast<int32_t>(_listItems.size() * SCROLLABLE_ROW_HEIGHT);
-}
-
-/**
- *
- *  rct2: 0x006AB0B6
- */
-static void WindowEditorObjectSelectionScrollMousedown(rct_window* w, int32_t scrollIndex, const ScreenCoordsXY& screenCoords)
-{
-    // Used for in-game object selection cheat to prevent crashing the game
-    // when windows attempt to draw objects that don't exist any more
-    window_close_all_except_class(WC_EDITOR_OBJECT_SELECTION);
-
-    int32_t selected_object = GetObjectFromObjectSelection(GetSelectedObjectType(w), screenCoords.y);
-    if (selected_object == -1)
-        return;
-
-    ObjectListItem* listItem = &_listItems[selected_object];
-    uint8_t object_selection_flags = *listItem->flags;
-    if (object_selection_flags & OBJECT_SELECTION_FLAG_6)
-        return;
-
-    w->Invalidate();
-
-    const CursorState* state = context_get_cursor_state();
-    OpenRCT2::Audio::Play(OpenRCT2::Audio::SoundId::Click1, 0, state->position.x);
-
-    if (gScreenFlags & SCREEN_FLAGS_TRACK_MANAGER)
+    /**
+     *
+     *  rct2: 0x006AB058
+     */
+    OpenRCT2String OnTooltip(const rct_widgetindex widgetIndex, const rct_string_id fallback) override
     {
-        if (!window_editor_object_selection_select_object(0, INPUT_FLAG_EDITOR_OBJECT_SELECT, listItem->repositoryItem))
-            return;
-
-        // Close any other open windows such as options/colour schemes to prevent a crash.
-        window_close_all();
-        // window_close(w);
-
-        // This function calls window_track_list_open
-        WindowEditorObjectSelectionManageTracks();
-        return;
+        if (widgetIndex >= WIDX_TAB_1 && static_cast<size_t>(widgetIndex) < WIDX_TAB_1 + std::size(ObjectSelectionPages))
+        {
+            auto ft = Formatter();
+            ft.Add<rct_string_id>(ObjectSelectionPages[(widgetIndex - WIDX_TAB_1)].Caption);
+            return { fallback, ft };
+        }
+        return { fallback, {} };
     }
 
-    int32_t flags = INPUT_FLAG_EDITOR_OBJECT_1 | INPUT_FLAG_EDITOR_OBJECT_SELECT_OBJECTS_IN_SCENERY_GROUP;
-    // If already selected
-    if (!(object_selection_flags & OBJECT_SELECTION_FLAG_SELECTED))
-        flags |= INPUT_FLAG_EDITOR_OBJECT_SELECT;
-
-    _maxObjectsWasHit = false;
-    if (!window_editor_object_selection_select_object(0, flags, listItem->repositoryItem))
+    void OnTextInput(rct_widgetindex widgetIndex, std::string_view text) override
     {
-        rct_string_id error_title = (flags & INPUT_FLAG_EDITOR_OBJECT_SELECT) ? STR_UNABLE_TO_SELECT_THIS_OBJECT
-                                                                              : STR_UNABLE_TO_DE_SELECT_THIS_OBJECT;
+        if (widgetIndex != WIDX_FILTER_TEXT_BOX || text.data() == nullptr)
+            return;
 
-        context_show_error(error_title, gGameCommandErrorText, {});
-        return;
-    }
+        if (strcmp(_filter_string, text.data()) == 0)
```

Comment:
This is unsafe to perform. You need to copy the string_view into a string to use strcmp.

Reply:
Done https://github.com/OpenRCT2/OpenRCT2/pull/16027/commits/de2d84eb142b8ae70b03ddf0d1215fe203df5e0d Also, is it proper etiquette to reply to each review with the commit URL? Sorry if it isn't and I'm spamming you

Reply:
Different people do it differently. Feel free to hit the resolve conversation button as you address each point. Then at a glance everyone can see if there is any outstanding.